compiler/testData/codegen/box/strings/stringFromJavaPlus.kt

```diff
@@ -0,0 +1,138 @@
+// IGNORE_BACKEND_FIR: JVM_IR
+// TARGET_BACKEND: JVM
+// WITH_RUNTIME
+
+// FILE: J.java
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class J {
+    public static String platformStringIsNull() {
+        return null;
+    }
+
+    @NotNull
+    public static String notNullStringIsNull() {
+        return null;
+    }
+
+    @Nullable
+    public static String nullableStringIsNull() {
+        return null;
+    }
+
+    @NotNull
+    public static String notNullStringIsNotNull() {
+        return "foo";
+    }
+
+    public static String platformStringIsNotNull() {
+        return "foo";
+    }
+
+    @Nullable
+    public static String nullableStringIsNotNull() {
+        return "foo";
+    }
+}
+
+// FILE: main.kt
```

Comment:
Minor: I usually give the "main" file in the multi-file test the same name as the test file (`stringFromJavaPlus.kt` in this case). Enables navigation from stack trace in IDEA.